omc3/model/model_creators/lhc_model_creator.py

```diff
@@ -330,13 +348,15 @@ def _get_madx_script_info_comments(self) -> str:
             )
         return info_comments
 
-    def _get_madx_main_sequence_loading(self) -> str:
+    def _get_madx_main_sequence_loading(self, acc_model_path: Path | None) -> str:
         accel: Lhc = self.accel
 
-        if accel.acc_model_path is not None:
-            main_call = f'call, file = \'{accel.acc_model_path / "lhc.seq"}\';'
-            if accel.year.startswith('hl'):
-                main_call += f'\ncall, file = \'{accel.acc_model_path / "hllhc_sequence.madx"}\';'
+        if acc_model_path is not None:
+            main_seq_rel = acc_model_path / "lhc.seq"
+            main_call = f"call, file = '{main_seq_rel}';"
+            if accel.year.startswith("hl"):
+                hllhc_rel = acc_model_path / "hllhc_sequence.madx"
+                main_call += f"\ncall, file = '{hllhc_rel}';"
             return main_call
```

Comment:
as far as I can see, you moved the `if self.acc_model_path is None` check out of this function, but only used the resulting variable in the function call, and you only call this function once with the acc_model_path. Seems a bit convoluted to me. What's the advantage over doing it in the function, as it was before?